### PR TITLE
Making a start on updating the iOS documentation

### PR DESCRIPTION
**Pull request by jackcolley**
Bringing iOS documentation up to date and providing a foundation to build on

### CODE REVIEW

File: src/flutter/codesigning.md [hunk 1]
```diff
@@ -1,25 +1,26 @@
 # Codesigning
 
-For the iOS component of Flutter apps, we use [Fastlane Match](https://docs.fastlane.tools/actions/match/) to manage signing certificates and provisioning profiles.
+For the iOS part of flutter apps, we use [Fastlane Match](https://docs.fastlane.tools/actions/match/) to manage signing certificates and provisioning profiles.
```
Comment (ptrbrynt) [inline src/flutter/codesigning.md:3]: Typo
```suggestion
For the iOS part of Flutter apps, we use [Fastlane Match](https://docs.fastlane.tools/actions/match/) to manage signing certificates and provisioning profiles.
```

File: src/flutter/codesigning.md [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
 
-Our [project templates](https://github.com/netsells/flutter_templates) are set up for this out-of-the-box, but below is a reference Matchfile. This should be added to the `ios` directory of your project.
+For each project we should create a new certificates repo that matches the main code repo naming conventions. e.g `my-project-certificates`
+
+This means that should the project ever need to be transferred or outsource engineers added to the repo, we don't have to expose all other projects certificates and profiles.
```
Comment (ptrbrynt) [inline src/flutter/codesigning.md:7]: This is a good idea - I'll update the Flutter template's `Matchfile`.

File: src/ios/README.md [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
-
-This means, if the current version is iOS 13, we support iOS 11 unless the project requires any specific functionality which is only included in later versions of iOS. 
-
-## Dependency management
-We use cocoapods for our dependancy management tool.
-
-## CI
-We use Bitrise for all of our CI needs. All new projects should be setup with Bitrise by default
+## 📋 Intro
+This is a living document that aims to document the agreed set of code standards for the iOS team. Please discuss any changes with the wider team before updating this document
```
Comment (ptrbrynt) [inline src/ios/README.md:2]:
```suggestion
This is a living document that aims to document the agreed set of code standards for the iOS team. Please discuss any changes with the wider team before updating this document.
```

File: src/ios/README.md [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
+We currently aim to support “current minus 2”.
+E.g If the current iOS version is iOS 15, then we aim to support iOS 13+. Any increases to this have to be a business lead decision.
+## 🧙 Third-Party Dependencies
+We try to utilise as few dependencies as possible. In some cases that is unavoidable. Where a dependency is used, it must be discussed with the team before adding it to the project. Where possible we use Swift Package Manager for this and fall back to Cocoapods if not.
+## 🖥 CI System
+We use Bitrise for our CI system with the following automations for branching conventions:
+release/* - Pushes/merges to these branches will trigger a beta build pointing at staging that uploads to Firebase App Distribution
+develop - Pushes/merges to this branch will trigger a build that runs the automated tests
+master - Pushes/merges to this branch will trigger a build pointing at production that uploads to App Store Connect
+Opening a PR - Opening a PR will trigger a build that runs the automated tests and code quality tools then reports back to Github
```
Comment (ptrbrynt) [inline src/ios/README.md:13]:
```suggestion
Opening a PR - Opening a PR will trigger a build that runs the automated tests and code quality tools then reports back to GitHub
```

File: src/ios/README.md [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
+## ✅ Pull Requests
+When creating a pull request, there are a few things to watch out for. Firstly, ensure that the base branch is set correctly.
+Once the branches are set, Github will populate the description with a pull request template. You should fill this template in with as much detail as possible for the reviewer.
```
Comment (ptrbrynt) [inline src/ios/README.md:16]:
```suggestion
Once the branches are set, GitHub will populate the description with a pull request template. You should fill this template in with as much detail as possible for the reviewer. 
```

File: src/ios/architecture.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# 🛠 Architecture Layers
+## 🍰 Layer Definitions
+### Data
+* The data layer contains the concrete implementations of the repository protocols from the domain layer
+* Repository implementations may depend on other classes such as data sources or services
+* It is a good idea (not a requirement) to use a Codable model as a DTO (Data Transfer Object) which represents the structure of the data returned from external services. These can then be converted to and from your entity struct. This means that your entities are not dependant on the data structure of the objects returned from external sources.
```
Comment (ptrbrynt) [inline src/ios/architecture.md:6]:
```suggestion
* It is a good idea (not a requirement) to use a Codable model as a DTO (Data Transfer Object) which represents the structure of the data returned from external services. These can then be converted to and from your entity struct. This means that your entities are not dependent on the data structure of the objects returned from external sources.
```

File: src/ios/architecture.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            * Pages
+                * My Bookings Page
+                    * MyBookingsPage.swift
+                    * MyBookingsPageViewModel.swift
+                * View Booking Page
+                    * ViewBookingPage.swift
+                    * ViewBookingPageViewModel.swift
+
+## 💯 Architecture Best Practices
+* A view’s state (@State) should only be used when absolutely necessary (e.g when building a form). Most of the time, it is far better to use the view model to hold and control all state.
```
Comment (ptrbrynt) [inline src/ios/architecture.md:61]:
```suggestion
* A view’s state (`@State`) should only be used when absolutely necessary (e.g when building a form). Most of the time, it is far better to use the view model to hold and control all state.
```

File: src/ios/bundler.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+eval_gemfile(plugins_path) if File.exist?(plugins_path)
+```
+
+Once you have created and populated the file in your project's directory. You can simply run bundle install
+
+## Using Bundler
+
+Ensure you have run `bundle install` successfully
+
+If you want to use cocoapods or fastlane, you simply need to prefix their commands with bundle exec
```
Comment (ptrbrynt) [inline src/ios/bundler.md:26]:
```suggestion
If you want to use Cocoapods or Fastlane, you simply need to prefix their commands with `bundle exec`
```

File: src/ios/testing.md [hunk 9]
```diff
@@ -1,9 +1,46 @@
-# Testing
+# 🧪 Testing
+Comprehensive automated tests are a crucial component of a healthy codebase. Not only do they allow behavious to be continuously verified as changes are made to a project. They also encourage good development habits, boost productivity and serve as living documentation for the intended functionality of your code.
```
Comment (ptrbrynt) [inline src/ios/testing.md:2]:
```suggestion
Comprehensive automated tests are a crucial component of a healthy codebase. Not only do they allow behaviours to be continuously verified as changes are made to a project; they also encourage good development habits, boost productivity and serve as living documentation for the intended functionality of your code.
```

File: src/ios/testing.md [hunk 10]
```diff
@@ -1,9 +1,46 @@
...
+* The goal of the test in unclear
+* The test has multiple reasons to fail
+* More than one thing is been tested
+* The test contains artificial delays
+* The test has hidden or internal dependencies
+* There are no assertions, expectations or verifications
+* The order in which a test is executed affects its results
+
+## 🪞 Matching the acceptance criteria
+We spend a lot of time in refinement sessions breaking down each ticket and writing the acceptance critieria for all functionality. Try to write your tests to match and verifiy the acceptance critera for the ticket you are working on.
```
Comment (ptrbrynt) [inline src/ios/testing.md:29]:
```suggestion
We spend a lot of time in refinement sessions breaking down each ticket and writing the acceptance critieria for all functionality. Try to write your tests to match and verify the acceptance critera for the ticket you are working on. 
```

File: src/ios/testing.md [hunk 11]
```diff
@@ -1,9 +1,46 @@
...
+
+## 🔬 Review tests in PRs carefully
+Its important that tests are reviewed carefully during the PR stage. When reviewing tests, look out for these red flags:
+* Tests with no assertions, expectations or verifications
+* Tests which have obviously been written to just satisfy the requirement for tests to be added
+* Not all logical branches are tested e.g only the happy path is tested
+
+## ⌛️ Don’t skip tests to save time
+Skipping tests today might save you time, but you are exchanging the saved time for technical debt. An untested codebase can’t scale efficiently, and will beocme more and more difficult to work on over time. The time you might save by not writing a test will need to be paid back - with significant interest later down the line.
+Its ok to be firm on this principle when communicating with non-technical staff. Ticket estimates should and must include sufficient allowance for engineers to write high-quality, well-tested code.
```
Comment (ptrbrynt) [inline src/ios/testing.md:46]:
```suggestion
Its okay to be firm on this principle when communicating with non-technical staff. Ticket estimates should and must include sufficient allowance for engineers to write high-quality, well-tested code.
```

File: src/ios/testing.md [hunk 12]
```diff
@@ -1,9 +1,46 @@
...
+When its time to open up a pull request, you should ensure you’re meeting the requirement that all tests are passing and new functionality is covered by tests.
+
+## 🔬 Review tests in PRs carefully
+Its important that tests are reviewed carefully during the PR stage. When reviewing tests, look out for these red flags:
+* Tests with no assertions, expectations or verifications
+* Tests which have obviously been written to just satisfy the requirement for tests to be added
+* Not all logical branches are tested e.g only the happy path is tested
+
+## ⌛️ Don’t skip tests to save time
+Skipping tests today might save you time, but you are exchanging the saved time for technical debt. An untested codebase can’t scale efficiently, and will beocme more and more difficult to work on over time. The time you might save by not writing a test will need to be paid back - with significant interest later down the line.
```
Comment (ptrbrynt) [inline src/ios/testing.md:45]:
```suggestion
Skipping tests today might save you time, but you are exchanging the saved time for technical debt. An untested codebase can’t scale efficiently, and will become more and more difficult to work on over time. The time you might save by not writing a test will need to be paid back - with significant interest later down the line.
```

**Review by ptrbrynt — CHANGES_REQUESTED** [review]
Really good stuff here! Just a few typos and spelling things but overall looking great 👍🏻

**Comment by jackcolley** [comment]
Thanks for the feedback @ptrbrynt - I've updated

**Review by ptrbrynt — APPROVED** [review]

**Pull request merged**